cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,290 @@
+    which can be used to dynamically convert logical resource names in pod specifications into actual FPGA
+    resource names, as advertised by the device plugin.
+
+    The webhook can also set environment variables to instruct the CRI-O prestart hook to program the FPGA
+    before launching the container.
+
+-   [FPGA CRI-O prestart hook](../fpga_crihook/README.md)
+
+    A [CRI-O](https://github.com/cri-o/cri-o) prestart hook that, upon instruction from the FPGA admission
+    controller, allocates and programs the FPGA before the container is launched.
+
+The repository also contains an [FPGA helper tool](../fpga_tool/README.md) that may be useful during
+development, initial deployment and debugging.
+
+# FPGA modes
+
+The FPGA plugin set can run in one of two modes:
+
+- `af`/`preprogrammed` mode, where the FPGA bitstreams are already loaded
+  onto the FPGA, and the plugins discover and advertises the existing
+  Accelerator Functions (AF).
+- `region`/`orchestration` mode, where the plugins locate and advertise
+  regions of the FPGA, and facilitate programing of those regions with the
+  requested bistreams.
+
+The example YAML deployments described in this document only currently support
+`af`/`preprogrammed` mode. To utilise `region`/`orchestrated` mode, either modify
+the existing YAML appropriately, or deploy 'by hand'.
+
+An overview diagram of `preprogrammed` mode is below. A diagram of `orchestrated` mode
+is provided in [an appendix](#orchestrated-mode).
+
+![Overview of `preprogrammed` mode](pictures/FPGA-preprogrammed.png)
+
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+Components can generally be installed either using DaemonSets or running them
+'by hand' on each node.
+
+## Pre-built images
+
+Pre-built images of the components are available on the [Docker hub](https://hub.docker.com/u/intel).
+These images are automatically built and uploaded to the hub from the latest `master` branch of
+this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers (of the form `x.y.z`, matching the branch/tag release number in this repo).
+
+The deployment YAML files supplied with these components in this repository use the images with the
+`devel` tag by default. If you do not build your own local images, then your Kubernetes cluster may
+pull down the `devel` images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the YAML deployment files appropriately.
+
+The following images are available on the Docker hub:
+
+- [The FPGA plugin](https://hub.docker.com/r/intel/intel-fpga-plugin)
+- [The FPGA admisson webhook](https://hub.docker.com/r/intel/intel-fpga-admissionwebhook)
+- [The FPGA CRI-O plugin (in the `initcontainer` image)](https://hub.docker.com/r/intel/intel-fpga-initcontainer)
+
```

Comment:
"The FPGA CRI-O prestart hook" should be used here.

Reply:
done